crates/rust-analyzer/src/cli/rustc_tests.rs

```diff
@@ -226,6 +226,7 @@ const SUPPORTED_DIAGNOSTICS: &[DiagnosticCode] = &[
 ];
 
 impl flags::RustcTests {
+    #[allow(clippy::redundant_locals)]
```

Comment:
This should probably be fixed instead of allowed

Reply:
I am unsure how. The relevant line is https://github.com/rust-lang/rust-analyzer/blob/1fdf3063d065bfff48a610b9bf656267cc934b1a/crates/rust-analyzer/src/cli/rustc_tests.rs#L249

If I remove it I get following error:

```
error[E0277]: the type `&mut Tester` may not be safely transferred across an unwind boundary
   --> crates/rust-analyzer/src/cli/rustc_tests.rs:243:54
    |
243 |                if let Err(e) = std::panic::catch_unwind({
    |  ______________________________------------------------_^
    | |                              |
    | |                              required by a bound introduced by this call
244 | |                  let tester = AssertUnwindSafe(&mut tester);
245 | |                  let p = p.clone();
246 | |                  move || {
    | |                  -------
    | |                  |
    | | _________________within this `{closure@crates/rust-analyzer/src/cli/rustc_tests.rs:246:17: 246:24}`
    | ||
247 | ||                     let _guard = stdx::panic_context::enter(p.display().to_string());
248 | ||                     tester.0.test(p);
249 | ||                 }
    | ||_________________- this tail expression is of type `{closure@rustc_tests.rs:246:17}`
250 | |              }) {
    | |______________^ `&mut Tester` may not be safely transferred across an unwind boundary
    |
    = help: within `{closure@crates/rust-analyzer/src/cli/rustc_tests.rs:246:17: 246:24}`, the trait `UnwindSafe` is not implemented for `&mut Tester`
    = note: `UnwindSafe` is implemented for `&Tester`, but not for `&mut Tester`
note: required because it's used within this closure
   --> crates/rust-analyzer/src/cli/rustc_tests.rs:246:17
    |
246 |                 move || {
    |                 ^^^^^^^
note: required by a bound in `catch_unwind`
   --> /home/urhengulas/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/panic.rs:141:40
    |
141 | pub fn catch_unwind<F: FnOnce() -> R + UnwindSafe, R>(f: F) -> Result<R> {
    |                                        ^^^^^^^^^^ required by this bound in `catch_unwind`
```

Reply:
In any case I think we can fix this in a follow-up PR

Reply:
That's a clippy bug then. We can fix this anyways by doing
```diff
                let tester = AssertUnwindSafe(&mut tester);
                let p = p.clone();
                move || {
                    let _guard = stdx::panic_context::enter(p.display().to_string());
-                    let tester = tester;
-                    tester.0.test(p);
+                    { tester }.0.test(p);
                }
 ```
 That should achieve the same thing as the let binding (forcing a move of the entire local)

Reply:
But let's do that in the follow up then

Reply:
Done https://github.com/rust-lang/rust-analyzer/pull/16404/commits/9e837790d14d114781aca386b9a4f2cdc85c3303